docs/USAGE-GUIDE.md

```diff
@@ -77,6 +77,28 @@ And when invoking CMake for your project, do one of two things:
  1. Set the `CMAKE_INSTALL_PREFIX` variable with the path to your s2n-tls build.
  2. If you have globally installed s2n-tls, do nothing, it will automatically be found.
 
+
+### Cross Compiling for 32 Bit Platforms
```

Comment:
Very strongly opposed to removing the documentation. If this is in our codebase it should be documented. 

If we want to send a "32 bit support is on shaky ground" message I think we should explicitly label it as such, rather than using using the absence of documentation to imply a lack of support.

Proposed additional message
> s2n-tls has support for 32 bit builds, but they are not built or tested in our CI, so they may or may not work. If you run into any issue let us know! We welcome pull requests and/or suggestions about how to better support 32 bit customers!

The is largely copied from the Rust Platform "Tier 3" support category wording: https://doc.rust-lang.org/beta/rustc/platform-support.html#tier-3

Once we have a build running in CI we would change this wording to reflect the Tier 2 status.

Reply:
Ok yeah, that makes sense. The additional message seems good to me. I just didn't want customers to read this and assume 32 bit is fully supported/tested.